src/mesh/mesh_communication_global_indices.C

```diff
@@ -841,6 +841,8 @@ void MeshCommunication::find_global_indices (const Parallel::Communicator & comm
     (processor_id_type, const std::vector<Parallel::DofObjectKey> & keys,
      std::vector<dof_id_type> & global_ids)
     {
+      // Avoid warnings about unused lambda captures in optimized builds.
+      libmesh_ignore(bbox);
```

Comment:
Shouldn't we just move bbox into the ```#ifndef NDEBUG``` block?

Reply:
That would be fine with me. IIRC `#ifndef NDEBUG` is equivalent to `#ifdef DEBUG` when we're building libmesh...

Reply:
Thanks!

> IIRC ```#ifndef NDEBUG``` is equivalent to ```#ifdef DEBUG``` when we're building libmesh...

Nope.  NDEBUG is turned off in devel mode; DEBUG isn't turned on until dbg mode.  This is actually super important because removing NDEBUG merely turns on all our constant-time-cost assertions while DEBUG turns on all our asymptotically-slower tests too.  So devel mode is still appropriate for debugging arbitrarily large problems while dbg mode is just intended to make debugging as easy as possible after you've gotten down to a cheap-enough test case.

Reply:
Hmm... then I think `bbox` is still an unused variable in devel mode in the current version of the code, as it is only used in an `#ifdef DEBUG` section.

Reply:
Ah, hell.  Can you check?  I don't have a recent clang version handy.

Reply:
Yep, definitely back so I'm going to go back to the `libmesh_ignore()` strategy.